### PR TITLE
Initial work for removing the iwdp dependency

### PR DESCRIPTION
**Pull request by umutuzgur**
* Add APIs for sending and receiving messages
* Add support for WIRFinalMessageKey and WIRPartialMessageKey
* Add a basic test

### Comments
* The send and receive functions are not fitting for the rpc case. We need to talk about how it works and modify accordingly
* I couldn't find a case where WIRPartialMessageKey is sent. This needs to be test when the service is used with appium-remote-debugger. It shouldn't be then that hard to fix it

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { default as Usbmux } from './lib/usbmux';
-import { getConnectedDevices, getOSVersion, startLockdownSession, startSyslogService, connectPort } from './lib/utilities';
+import { getConnectedDevices, getOSVersion, startLockdownSession, startSyslogService, startWebInspectorService, connectPort } from './lib/utilities';
```
Comment (mykola-mokhnach) [inline index.js]: Please wrap long lines

File: lib/webinspector/index.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import LengthBasedSplitter from '../util/transformer/length-based-splitter';
+
+const WEB_INSPECTOR_SERVICE_NAME = 'com.apple.webinspector';
+
+class WebInspectorService {
+
+  constructor (socketClient) {
+    this._socketClient = socketClient;
+
+    this._decoder = new WebInspectorDecoder(1000000);
```
Comment (mykola-mokhnach) [inline lib/webinspector/index.js]: What is this magic number?

File: lib/webinspector/transformer/webinspector-decoder.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
+/* eslint-disable promise/prefer-await-to-callbacks */
+import Stream from 'stream';
+import { parse } from '../../util/plist';
```
Comment (mykola-mokhnach) [inline lib/webinspector/transformer/webinspector-decoder.js]: I find the name of this function too generic

File: lib/webinspector/transformer/webinspector-encoder.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+class WebInspectorEncoder extends Stream.Transform {
+
+  constructor () {
+    super({ objectMode: true });
+  }
+
+  _transform (data, encoding, callback) {
+    let payloadBuffer = createPlist(data, true);
+    for (let i = 0; i < payloadBuffer.length; i += this._encode(payloadBuffer, 0)) {}
```
Comment (mykola-mokhnach) [inline lib/webinspector/transformer/webinspector-encoder.js]: for(); should be enough

File: test/webinspector/webinspector-service-specs.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    webInspectorService.close();
+    if (server) {
+      server.close();
+    }
+  });
+
+  it('should receive webinspector WIRFinalMessageKey messages back', async function () {
+    ({server, socket} = await getServerWithFixtures(fixtures.WEBINSPECTOR_MESSAGES));
+    webInspectorService = new WebInspectorService(socket);
+    let obj = {'__argument': {'WIRConnectionIdentifierKey': '990cc163-d8b2-4d22-8d1c-644e100a5a07'}, '__selector': '_rpc_reportIdentifier:'};
```
Comment (mykola-mokhnach) [inline test/webinspector/webinspector-service-specs.js]: do we need to quote the key?

File: test/webinspector/webinspector-service-specs.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+chai.should();
+
+describe('webinspector', function () {
+  let server;
+  let socket;
+  let webInspectorService;
+
+  afterEach(function () {
+    webInspectorService.close();
```
Comment (mykola-mokhnach) [inline test/webinspector/webinspector-service-specs.js]: should we check first if these are defined?

File: lib/webinspector/transformer/webinspector-encoder.js (hunk 4 above)
Comment (umutuzgur) [reply lib/webinspector/transformer/webinspector-encoder.js]: I actually have a bug there. It should be `this._encode(payloadBuffer, i)`

File: lib/webinspector/transformer/webinspector-decoder.js (hunk 3 above)
Comment (umutuzgur) [reply lib/webinspector/transformer/webinspector-decoder.js]: I can change it when I move it to `appium-support`

**Pull request merged**